### ISSUE TITLE
JDK16 (aarch64_linux) cmdLineTester Segmentation error vmState=0x00000000 at libj9jit29.so

### ISSUE THREAD

**Issue by JasonFengJ9** [eclipse/openj9#12006]
Failure link
------------

From an internal build `Test_openjdk16_j9_sanity.functional_aarch64_linux/6`:
```
02:35:24  openjdk version "16" 2021-03-16
02:35:24  OpenJDK Runtime Environment AdoptOpenJDK (build 16+36-202102180509)
02:35:24  Eclipse OpenJ9 VM AdoptOpenJDK (build master-8d60d1d11, JRE 16 Linux aarch64-64-Bit Compressed References 20210218_32 (JIT enabled, AOT enabled)
02:35:24  OpenJ9   - 8d60d1d11
02:35:24  OMR      - 57056f99c
02:35:24  JCL      - 197706b451 based on jdk-16+36)
```
to rebuild the failed tests in <Jenkins URL>=https://hyc-runtimes-jenkins.swg-devops.com/job/Grinder, use the following links:
12:35:09  https://hyc-runtimes-jenkins.swg-devops.com/job/Grinder/parambuild/?JDK_VERSION=16&JDK_IMPL=openj9&JDK_VENDOR=adoptopenjdk&BUILD_LIST=functional&PLATFORM=aarch64_linux_mixed&TARGET=SharedCPEntryInvokerTests_0
12:35:09  https://hyc-runtimes-jenkins.swg-devops.com/job/Grinder/parambuild/?JDK_VERSION=16&JDK_IMPL=openj9&JDK_VENDOR=adoptopenjdk&BUILD_LIST=functional&PLATFORM=aarch64_linux_mixed&TARGET=cmdLineTester_ShareClassesSimpleSanity_0
12:35:09  https://hyc-runtimes-jenkins.swg-devops.com/job/Grinder/parambuild/?JDK_VERSION=16&JDK_IMPL=openj9&JDK_VENDOR=adoptopenjdk&BUILD_LIST=functional&PLATFORM=aarch64_linux_mixed&TARGET=cmdLineTester_SCURLHelperTests_90_0

Optional info
-------------

Failure output (captured from console output)
---------------------------------------------
```
===============================================
Running test cmdLineTester_ShareClassesSimpleSanity_0 ...
===============================================
cmdLineTester_ShareClassesSimpleSanity_0 Start Time: Thu Feb 18 15:25:01 2021 Epoch Time (ms): 1613661901564
variation: Mode110
JVM_OPTIONS:  -Xjit -Xgcpolicy:gencon -Xnocompressedrefs 
Testing: Simple Sanity Run and Use Cache
Test start time: 2021/02/18 15:25:03 Greenwich Mean Time
Running command: "/home/jenkins/workspace/Test_openjdk16_j9_sanity.functional_aarch64_linux_testList_0/openjdkbinary/j2sdk-image/bin/java"  -Xjit -Xgcpolicy:gencon -Xnocompressedrefs  -Xshareclasses:name=HelloWorldSanity -cp /home/jenkins/workspace/Test_openjdk16_j9_sanity.functional_aarch64_linux_testList_0/openjdk-tests/TKG/../../jvmtest/functional/cmdLineTests/utils/utils.jar VMBench.FibBench
Time spent starting: 6 milliseconds
Time spent executing: 3259 milliseconds
Test result: FAILED
 [ERR] Unhandled exception
 [ERR] Type=Segmentation error vmState=0x00000000
 [ERR] J9Generic_Signal_Number=00000018 Signal_Number=0000000b Error_Value=00000000 Signal_Code=00000001
 [ERR] Handler1=0000FFFF98EBBF38 Handler2=0000FFFF98E333E8 InaccessibleAddress=00000000941E1360
 [ERR] R0=0000FFFF940DB400 R1=0000FFFE7BB71344 R2=0000FFFE7BB71860 R3=0000FFFF99D38A60
 [ERR] R4=0000000000000000 R5=0000FFFF9423F3E0 R6=0000FFFF93574500 R7=0000FFFE7BB71848
 [ERR] R8=0000FFFE7BB7129C R9=0000FFFE7BB6B0FC R10=0000FFFE7BB71344 R11=0000FFFE7BB71810
 [ERR] R12=000000000000FFC0 R13=0000FFFF940FCFE0 R14=FFFFFFFFFFFFFFF8 R15=0000FFFF99D38BE0
 [ERR] R16=0000FFFF93998A00 R17=0000FFFF99DEFF80 R18=0000FFFF93596728 R19=0000FFFF940DB400
 [ERR] R20=00000000941E1300 R21=0000FFFF93552990 R22=0000FFFF93558930 R23=0000FFFF93574500
 [ERR] R24=0000FFFF93558E30 R25=0000FFFF93558420 R26=0000FFFF93581900 R27=0000000000000182
 [ERR] R28=0000000000000001 R29=0000FFFF99D38990 R30=0000FFFF93D8397C R31=0000FFFF99D38990
 [ERR] PC=0000FFFF93D8F7C4 SP=0000FFFF99D38990 PSTATE=0000000080000000
 [ERR] V0 0000ffff9423f3b8 (f: 2485384192.000000, d: 1.390662e-309)
 [ERR] V1 000000003e3ab283 (f: 1044034176.000000, d: 5.158214e-315)
 [ERR] V2 3eb829560eaab070 (f: 246067312.000000, d: 1.440136e-06)
 [ERR] V3 bfdff97086029423 (f: 2248315904.000000, d: -4.995996e-01)
 [ERR] V4 bfd00ea348b88334 (f: 1220051712.000000, d: -2.508934e-01)
 [ERR] V5 3fd5575b0be00b6a (f: 199232368.000000, d: 3.334568e-01)
 [ERR] V6 bfdffffef20a4123 (f: 4060758272.000000, d: -4.999997e-01)
 [ERR] V7 3fe62e42fefa39ef (f: 4277811712.000000, d: 6.931472e-01)
 [ERR] V8 0000000000000000 (f: 0.000000, d: 0.000000e+00)
 [ERR] V9 0000000000000000 (f: 0.000000, d: 0.000000e+00)
 [ERR] V10 0000000000000000 (f: 0.000000, d: 0.000000e+00)
 [ERR] V11 0000000000000000 (f: 0.000000, d: 0.000000e+00)
 [ERR] V12 0000000000000000 (f: 0.000000, d: 0.000000e+00)
 [ERR] V13 0000000000000000 (f: 0.000000, d: 0.000000e+00)
 [ERR] V14 0000000000000000 (f: 0.000000, d: 0.000000e+00)
 [ERR] V15 0000000000000000 (f: 0.000000, d: 0.000000e+00)
 [ERR] V16 bff0000000000000 (f: 0.000000, d: -1.000000e+00)
 [ERR] V17 0060000200000000 (f: 0.000000, d: 7.120250e-307)
 [ERR] V18 0000000000000002 (f: 2.000000, d: 9.881313e-324)
 [ERR] V19 0000ffff93998bf8 (f: 2476313600.000000, d: 1.390662e-309)
 [ERR] V20 0000000000000000 (f: 0.000000, d: 0.000000e+00)
 [ERR] V21 0000000000000000 (f: 0.000000, d: 0.000000e+00)
 [ERR] V22 0000000000000000 (f: 0.000000, d: 0.000000e+00)
 [ERR] V23 0000000000000000 (f: 0.000000, d: 0.000000e+00)
 [ERR] V24 0000000000000000 (f: 0.000000, d: 0.000000e+00)
 [ERR] V25 0000000000000000 (f: 0.000000, d: 0.000000e+00)
 [ERR] V26 0000000000000000 (f: 0.000000, d: 0.000000e+00)
 [ERR] V27 0000000000000000 (f: 0.000000, d: 0.000000e+00)
 [ERR] V28 0000000000000000 (f: 0.000000, d: 0.000000e+00)
 [ERR] V29 0000000000000000 (f: 0.000000, d: 0.000000e+00)
 [ERR] V30 0000000000000000 (f: 0.000000, d: 0.000000e+00)
 [ERR] V31 0000000000000000 (f: 0.000000, d: 0.000000e+00)
 [ERR] Module=/home/jenkins/workspace/Test_openjdk16_j9_sanity.functional_aarch64_linux_testList_0/openjdkbinary/j2sdk-image/lib/default/libj9jit29.so
 [ERR] Module_base_address=0000FFFF93633000
 [ERR] Target=2_90_20210218_32 (Linux 4.15.0-132-generic)
 [ERR] CPU=aarch64 (8 logical CPUs) (0x3e8127000 RAM)
 [ERR] ----------- Stack Backtrace -----------
 [ERR] (0x0000FFFF98E30040 [libj9prt29.so+0x26040])
 [ERR] ---------------------------------------
 [ERR] JVMDUMP039I Processing dump event "gpf", detail "" at 2021/02/18 15:25:03 - please wait.
 [ERR] JVMDUMP032I JVM requested System dump using '/home/jenkins/workspace/Test_openjdk16_j9_sanity.functional_aarch64_linux_testList_0/openjdk-tests/TKG/output_16136609827891/cmdLineTester_ShareClassesSimpleSanity_0/core.20210218.152503.27212.0001.dmp' in response to an event
 [ERR] JVMPORT030W /proc/sys/kernel/core_pattern setting "|/usr/share/apport/apport %p %s %c %d %P %E" specifies that the core dump is to be piped to an external program.  Attempting to rename either core or core.27229.
 [ERR] 
 [ERR] JVMDUMP010I System dump written to /home/jenkins/workspace/Test_openjdk16_j9_sanity.functional_aarch64_linux_testList_0/openjdk-tests/TKG/output_16136609827891/cmdLineTester_ShareClassesSimpleSanity_0/core.20210218.152503.27212.0001.dmp
 [ERR] JVMDUMP032I JVM requested Java dump using '/home/jenkins/workspace/Test_openjdk16_j9_sanity.functional_aarch64_linux_testList_0/openjdk-tests/TKG/output_16136609827891/cmdLineTester_ShareClassesSimpleSanity_0/javacore.20210218.152503.27212.0002.txt' in response to an event
 [ERR] JVMDUMP010I Java dump written to /home/jenkins/workspace/Test_openjdk16_j9_sanity.functional_aarch64_linux_testList_0/openjdk-tests/TKG/output_16136609827891/cmdLineTester_ShareClassesSimpleSanity_0/javacore.20210218.152503.27212.0002.txt
 [ERR] JVMDUMP032I JVM requested Snap dump using '/home/jenkins/workspace/Test_openjdk16_j9_sanity.functional_aarch64_linux_testList_0/openjdk-tests/TKG/output_16136609827891/cmdLineTester_ShareClassesSimpleSanity_0/Snap.20210218.152503.27212.0003.trc' in response to an event
 [ERR] JVMDUMP010I Snap dump written to /home/jenkins/workspace/Test_openjdk16_j9_sanity.functional_aarch64_linux_testList_0/openjdk-tests/TKG/output_16136609827891/cmdLineTester_ShareClassesSimpleSanity_0/Snap.20210218.152503.27212.0003.trc
 [ERR] JVMDUMP032I JVM requested JIT dump using '/home/jenkins/workspace/Test_openjdk16_j9_sanity.functional_aarch64_linux_testList_0/openjdk-tests/TKG/output_16136609827891/cmdLineTester_ShareClassesSimpleSanity_0/jitdump.20210218.152503.27212.0004.dmp' in response to an event
 [ERR] JVMDUMP013I Processed dump event "gpf", detail "".
>> Success condition was not found: [Output match: Fibonacci: iterations = 10000]
>> Failure condition was not found: [Output match: Error:]
>> Failure condition was found: [Output match: Unhandled Exception]
>> Failure condition was not found: [Output match: Exception:]

cmdLineTester_ShareClassesSimpleSanity_0_FAILED
```

Similar Segmentation error vmState=0x00000000 at libj9jit29.so for:
```
cmdLineTester_SCURLHelperTests_90_0_FAILED
cmdLineTester_SCHelperCompatTests_unix_0_FAILED
cmdLineTester_SCCommandLineOptionTests_0_FAILED
...
```

**Comment by pshipton** [comment]
Is this mixedrefs? Seems like it has to be. -Xnocompressedrefs problem in aarch64 builds are being resolved by https://github.com/eclipse/openj9/pull/11987

**Comment by JasonFengJ9** [comment]
> Is this mixedrefs?

Yeah, it is as per the rebuild link: https://hyc-runtimes-jenkins.swg-devops.com/job/Grinder/parambuild/?JDK_VERSION=16&JDK_IMPL=openj9&JDK_VENDOR=adoptopenjdk&BUILD_LIST=functional&PLATFORM=aarch64_linux_mixed&TARGET=SharedCPEntryInvokerTests_0

**Comment by pshipton** [comment]
Re-opened to fix in the 0.25.0 release branch.

**Comment by vij-singh** [comment]
@0xdaryl @knn-k We'll need to deliver this to the 0.25.0 branch.
https://github.com/eclipse/openj9/pull/11987

**Comment by knn-k** [comment]
I merged #12109 to the 0.25.0 branch.
